### ISSUE TITLE
EDTF datatype append_search_filter returning fuzzy results

### ISSUE THREAD

**Issue by aj-he** [archesproject/arches#10112]
**INVITATION FOR COMMENT**

We have found that when searching using advanced search against a edtf node, the returned results are not always adhering to the required time range and provides a "fuzziness" to the results.

Through experimentation it was found that the EDTF datatype's append_search_filter function used "should" rather than "must" as all other datatypes use. Instead, using "must", or wrapping the two "should" filters in a Bool query that itself was added as a "must" greatly increase the accuracy.

The behaviour observed seems to suggest that an OR operation is being used alongside an AND and adding more results as a consequence.

This has only been tested in 6.1 and 6.2, but the affected function appears the same in v7 so assuming it likely has the same behaviour.

I have added the "must" solution below and **invite comment** on whether it should be changed to this or if the original "should" only approach was done for a specific reason. Does anyone see the same issue?

![image](https://github.com/archesproject/arches/assets/25295526/63711a68-7aa5-4f37-86ec-22e495b7d906)

**Comment by aj-he** [comment]
@chiatt, @apeters - let me know if you have any thoughts on this. I'll want to look at submitting a PR in the next few weeks.

**Comment by apeters** [comment]
Hi @aj-he,
I have not been able to replicate the issue you bring up.  Is there a specific set of data and search that can highlight this issue?

Also, a couple of observations:
On line 418 in the image above, changing "should" to must shouldn't have any affect, as we have a [minimum should match property set to 1](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html#bool-min-should-match).  Because of that, this effectively becomes a "must" search.  I found a different issue related to that search and filed a ticket [here](https://github.com/archesproject/arches/issues/10114).

RE: lines 434-438 in the image above.  Adding a wrapping boolean clause as you have done doesn't seem to change the search appreciably (at least as far as I can test) as further up the query this boolean clause is ultimately wrapped in a "must" clause.  See below:
```
"bool": {
    "filter": [],
    "must": [
        {
            "nested": {
                "path": "tiles",
                "query": {
                    "bool": {
                        "filter": [],
                        "minimum_should_match": 1,
                        "must": [],
                        "must_not": [],
                        "should": [
                            {
                                "range": {
                                    "tiles.data.b74e29fe-c8c4-11ed-aab4-0242ac130009.dates.date": {
                                        "gte": 19710805,
                                        "lte": 19710805
                                    }
                                }
                            },
                            {
                                "range": {
                                    "tiles.data.b74e29fe-c8c4-11ed-aab4-0242ac130009.date_ranges.date_range": {
                                        "gte": 19710805,
                                        "lte": 19710805,
                                        "relation": "intersects"
                                    }
                                }
                            }
                        ]
                    }
                }
            }
        },
        {
            "bool": {
                "filter": [],
                "must": [],
                "must_not": [],
                "should": []
            }
        }
    ],
    "must_not": [],
    "should": []
}
```
One thing I know we can't do is change the "should" queries to "must" (see image below) because there are times when only an exact date is saved and no "data_range" value is indexed.  In this case using a must clause on that will fail to return that document, even though it should.
<img width="1041" alt="Screenshot 2023-10-09 at 2 36 33 PM" src="https://github.com/archesproject/arches/assets/823773/8fd53453-0258-4124-801b-4985302a87b0">

**Comment by aj-he** [comment]
@apeters I'll see if I can get an example lined up to show the variation we see.

**Comment by aj-he** [comment]
Hi @apeters, I've compiled an example where we can see the issue. Let me know if you want to arrange an in person demo.

TLDR; - if we have 2 EDTF nodes that represent a start and end date within one nodegroup, using both in advanced search will return resources that are within those dates, but also resources that are "close".

Our internal Warden application is being migrated from 5.2 to 6.2. It has ~900,000 resources. 

Search the monuments model for resources who have a construction phase (card) that starts after the "From Date" and ends before the "To Date", and with a monument type of "Church".

![criteria](https://github.com/archesproject/arches/assets/25295526/d6d19f52-df89-45d2-bdea-8721cd8da892)

![show affect of fix](https://github.com/archesproject/arches/assets/25295526/a17efe3f-18d0-43e7-82ad-db76e5061183)

**Out of the box we get we get 6462 results**. Looking at the first few records, it looks correct. Towards the end, there are records that don't meet all the requirements. Type might be Church but the dates may slightly overlap the From-To dates, or are close but don't intersect.

![close match](https://github.com/archesproject/arches/assets/25295526/a297e8ec-f3eb-4a88-a57c-99731b407f65)


**With the fix, we see that there are 449 results** and all resources we sample has a construction phase card that exactly meet the search criteria.

![exact match](https://github.com/archesproject/arches/assets/25295526/3e17875f-9a2b-4e57-ac4a-dc6a48dbd9e6)

**Comment by apeters** [comment]
Hi @aj-he, 
So I did confirm that when you have more than one EDTF node in a card that using a `should` bool query returns extra results because only 1 of the `should` operators needs to pass.  So, that's a problem.  Your fix (see below) to the issue also seems to work well and does indeed fix the issue.   Good catch!  I'd be happy to review a PR when submit one!

```
group_query = Bool()
group_query.should(Range(field="tiles.data.%s.dates.date" % (str(node.pk)), **operators))
group_query.should(Range(field="tiles.data.%s.date_ranges.date_range" % (str(node.pk)), relation="intersects", **operators))
query.must(group_query)
```

**Comment by aj-he** [comment]
Closed complete #10202
